src/main/kotlin/com/pubnub/api/subscribe/eventengine/effect/EmitMessagesProvider.kt

```diff
@@ -0,0 +1,25 @@
+package com.pubnub.api.subscribe.eventengine.effect
+
+import com.pubnub.api.models.consumer.PNStatus
+import com.pubnub.api.models.consumer.pubsub.PNMessageResult
+import com.pubnub.api.models.consumer.pubsub.PNPresenceEventResult
+import com.pubnub.api.models.consumer.pubsub.PNSignalResult
+import com.pubnub.api.models.consumer.pubsub.files.PNFileEventResult
+import com.pubnub.api.models.consumer.pubsub.message_actions.PNMessageActionResult
+import com.pubnub.api.models.consumer.pubsub.objects.PNObjectEventResult
+
+interface EmitMessagesProvider {
+    fun announce(status: PNStatus) // toDo move to EmitStatusProvider
+
+    fun announce(message: PNMessageResult)
+
+    fun announce(presence: PNPresenceEventResult) // toDo move to Presence EventEngine
+
+    fun announce(signal: PNSignalResult)
+
+    fun announce(messageAction: PNMessageActionResult)
+
+    fun announce(pnObjectEventResult: PNObjectEventResult)
+
+    fun announce(pnFileEventResult: PNFileEventResult)
```

Comment:
Another question one might have is do we actually want to have this distinction on method level? Do we need so many methods named the same or maybe just one would be enough? 

```
    fun betterMethodNameThenAnnounceMaybe(event: PNEvent)
```

Reply:
I though that when putting embedding EE into SDK we could have ListenerManager implemented this interface. It has all those method implemented but doesn't have method announce(event: PNEvent)
What do you think?